lib/slack/real_time/config.rb

```diff
@@ -15,6 +18,23 @@ def reset
         self.websocket_ping = 30
         self.websocket_proxy = nil
         self.token = nil
+        self.concurrency = method(:detect_concurrency)
+      end
+
+      def concurrency
+        (val = @concurrency).respond_to?(:call) ? val.call : val
```

Comment:
Trying to be too clever. This actually doesn't save any cycles and is probably even slower than 

```ruby
@concurrency.respond_to?(:call) ? @concurrency.call : @concurrency
```